java/ql/src/Security/CWE/CWE-117/LogInjection.ql

```diff
@@ -0,0 +1,21 @@
+/**
+ * @name Log Injection
+ * @description Building log entries from user-controlled data may allow
+ *              insertion of forged log entries by malicious users.
+ * @kind path-problem
+ * @problem.severity error
+ * @security-severity 7.8
+ * @precision high
```

Comment:
Should either of these values be adjusted? You commented that maybe we should lower precision?

Reply:
Yes. `@security-severity` is calculated automatically, so *probably* we shouldn't change it. `@precision` is difficult to determine, because the query raises a high number of alerts in OSS projects, so it's a lot of work to check one by one whether they are TPs or not. Alternatively, we could reduce `@problem.severity` to `warning` too.

I want users to be aware of how "noisy" the query is, but I'm not sure which of those values we should tweak to reflect that.

Reply:
`@problem.severity` represents the seriousness of a TP, whereas `@precision` represents the expected TP/FP ratio. So if it's very noisy then `@precision` should be reduced to `medium` or perhaps even `low`.

Reply:
I don't expect this to have a significantly lower TP/FP ratio than other taint tracking queries, but it's the sheer amount of results (even if them all were TPs) which makes me want to flag this in some way. So I think `@precision low` would be too much. Let's go with `medium` if you agree.

Reply:
SGTM.

Reply:
```suggestion
 * @problem.severity error
 * @security-severity 7.8
 * @precision medium
```